### ISSUE TITLE
pyyaml/cython error when installing with poetry install

### ISSUE THREAD

**Issue by Iain-S** [alan-turing-institute/sqlsynthgen#120]
As of a recent release of cython v3, `poetry install` will fail with an error such as:

> • Installing python-dotenv (1.0.0)
  • Installing pyyaml (5.4.1): Failed
...
  ChefBuildError
...
raise AttributeError(attr)
  AttributeError: cython_sources
...
Note: This error originates from the build backend, and is likely not a problem with poetry but with pyyaml (5.4.1) not supporting PEP 517 builds. You can verify this by running 'pip wheel --use-pep517 "pyyaml (==5.4.1)"'.

We believe that this is because pyyaml 5.x.x cannot be built with the latest release of cython.
Although poetry locks the versions of dependencies, it seems that it does **not** lock the versions of build dependencies.

The main pyyaml issue is [here](https://github.com/yaml/pyyaml/issues/601) where the maintainer sounds unlikely to do a 5.4.2 release. We could update to pyyaml 6.x.x but that would require us to upgrade other dependencies and, ultimately, use sqlalchemy v2, which is currently incompatible with smartnoise SQL. See #104 for more.

**Comment by Iain-S** [comment]
A potential workaround is to manually install cython < 3 and use `pip`s `--no-build-isolation` option to control the version of cython used to build pyyaml:

```bash
cd sqlsynthgen
poetry run pip install cython<3
poetry run pip install wheel
poetry run pip install --no-build-isolation pyyaml==5.4.1
poetry install
```

Full credit to [this](https://github.com/yaml/pyyaml/pull/726#issuecomment-1641500321) comment for coming up with the workaround.

**Comment by Iain-S** [comment]
Closing this as we think it has been fixed by the upgrade to SQL Alchemy v2.

**Comment by cptanalatriste** [comment]
I've seen the same error when running `pip install sqlsynthgen` from a `Dockerfile`. Luckily, the workaround also fixed it in this case.

**Comment by cptanalatriste** [comment]
I think the workaround is the problem: installing `opendp`  in isolation shows no problem. And it depends on `pyyaml==6.0`:

```
jovyan@8bcfc2dffc6c:/workspaces/sqlsynthgen-health-demo$ python
Python 3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:47:07) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import opendp
>>> exit()
jovyan@8bcfc2dffc6c:/workspaces/sqlsynthgen-health-demo$ pip show opendp
Name: opendp
Version: 0.8.0
Summary: Python bindings for the OpenDP Library
Home-page: https://opendp.org
Author: The OpenDP Project
Author-email: info@opendp.org
License: UNKNOWN
Location: /opt/conda/lib/python3.10/site-packages
Requires: 
Required-by: 
jovyan@8bcfc2dffc6c:/workspaces/sqlsynthgen-health-demo$ pip show pyyaml
Name: PyYAML
Version: 6.0
Summary: YAML parser and emitter for Python
Home-page: https://pyyaml.org/
Author: Kirill Simonov
Author-email: xi@resolvent.net
License: MIT
Location: /opt/conda/lib/python3.10/site-packages
Requires: 
Required-by: bokeh, dask, distributed, jupyter-events
jovyan@8bcfc2dffc6c:/workspaces/sqlsynthgen-health-demo$ 
```

**Comment by mhauru** [comment]
Sorry I didn't fully follow. How can I reproduce the problem that comes up when you run `pip install sqlsynthgen`?

**Comment by cptanalatriste** [comment]
@mhauru , this is what I did on a Linux Docker container:

1. Install SSG using `pip install sqlsynthgen`
2. Fails, because of the Cython issue.
3. Use @Iain-S 's workaround, to get `pip install sqlsynthgen` working.
4. Now, when using `sqlsynthgen make-tables` you get an `opendp` error. I suspect the cause of this error is the workaround.

**Comment by mhauru** [comment]
Can reproduce. I think the issues is just that we haven't made a new pypi release in a long while. Whooops... Should definitely do that.

**Comment by mhauru** [comment]
Doing that now, see #164

**Comment by mhauru** [comment]
Now fixed by the release of 0.4.0. Thanks Carlos for spotting, this was a significant oversight to fix.
